lib/auth/default_current_user_provider.rb

```diff
@@ -86,17 +86,24 @@ def current_user
       api_key ||= request[API_KEY]
     end
 
-    auth_token = request.cookies[TOKEN_COOKIE] unless user_api_key || api_key
+    auth_cookie = request.cookies[TOKEN_COOKIE].presence unless user_api_key || api_key
+    if auth_cookie
+      begin
+        cookie = DiscourseAuthCookie.parse(auth_cookie)
+      rescue  DiscourseAuthCookie::InvalidCookie
+        cookie = nil
+      end
```

Comment:
This is great, because we can avoid the DB lookup for fake tokens 😍 

However, we are not checking any kind of timestamp here. A token from 10 years ago would still be 'valid' by this logic. I think we could make this much safer with just a little extra refactoring:

1. In the cookie, remove `valid_till`, and replace it with `issued_at`
2. In the request_tracker, replace `cookie.valid_till < Time.zone.now.to_i` with `cookie.issued_at > ROTATE_TIME.ago` (plus a little leeway, as described in my other comment)
3. Here in the default_current_user_provider, check `cookie.issued_at > SiteSetting.maximum_session_age.hours.ago`


What do you think? 🤔

Reply:
Sounds good to me! I've made this change let me know what you think.